File: 2trabajoConstructores/go/interfaz/animal_con_parametros.go
package main

import "fmt"

type AnimalConParametros struct {
	Nombre string
	Sonido string
}

func NuevoAnimalConParametros(nombre, sonido string) AnimalConParametros {
	return AnimalConParametros{
		Nombre: nombre,
		Sonido: sonido,
	}
}

func (a AnimalConParametros) HacerSonido() {
	fmt.Printf("%s hace: %s\n", a.Nombre, a.Sonido)
}

File: 2trabajoConstructores/js/clase normal/AnimalConParametros.js
class AnimalConParametros {
    constructor(nombre, sonido) {
        this.nombre = nombre;
        this.sonido = sonido;
    }

    hacerSonido() {
        console.log(`${this.nombre} hace: ${this.sonido}`);
    }
}

module.exports = AnimalConParametros;
